src/crimson/os/futurized_store.h

```diff
@@ -60,6 +66,50 @@ class FuturizedStore {
     uint64_t offset,
     size_t len,
     uint32_t op_flags = 0) = 0;
+
+  virtual seastar::future<CollectionRef> open_collection(const coll_t& cid) = 0;
```

Comment:
How about dropping open_collection/create_new_collection/do_transaction from FuturizedStore and use them through get_local_shard_store()?

Reply:
I think it is better keep its transform within futurized store, don't expose it to osd. otherwise each osd parts calling these interfaces need know it call shard store.

Reply:
Looks to me that all these duplicated interfaces (open_collection/create_new_collection/do_transaction/read) are all related to the meta-coll operations. Proper cleanups with OSDMeta to work with FuturizedShardStore can address the repeating redirections. Also it looks reasonable to access the local FuturizedShardStore instead for the meta-coll operations.

The point is that it is simpler/more-straiforward to make FuturizedStore/FuturizedShardStore interfaces disjoint.

Reply:
done in a new commit.